sdk/resources/Azure.ResourceManager.Resources/api/Azure.ResourceManager.Resources.net8.0.cs

```diff
@@ -1635,11 +1653,92 @@ protected virtual void JsonModelWriteCore(System.Text.Json.Utf8JsonWriter writer
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Resources.Models.DeploymentIdentity>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Resources.Models.DeploymentIdentity>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct DeploymentIdentityType : System.IEquatable<Azure.ResourceManager.Resources.Models.DeploymentIdentityType>
+    {
+        private readonly object _dummy;
+        private readonly int _dummyPrimitive;
+        public DeploymentIdentityType(string value) { throw null; }
+        public static Azure.ResourceManager.Resources.Models.DeploymentIdentityType None { get { throw null; } }
+        public static Azure.ResourceManager.Resources.Models.DeploymentIdentityType UserAssigned { get { throw null; } }
+        public bool Equals(Azure.ResourceManager.Resources.Models.DeploymentIdentityType other) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.ResourceManager.Resources.Models.DeploymentIdentityType left, Azure.ResourceManager.Resources.Models.DeploymentIdentityType right) { throw null; }
+        public static implicit operator Azure.ResourceManager.Resources.Models.DeploymentIdentityType (string value) { throw null; }
+        public static bool operator !=(Azure.ResourceManager.Resources.Models.DeploymentIdentityType left, Azure.ResourceManager.Resources.Models.DeploymentIdentityType right) { throw null; }
+        public override string ToString() { throw null; }
+    }
     public partial class DeploymentParameter : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Resources.Models.DeploymentParameter>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Resources.Models.DeploymentParameter>
     {
         public DeploymentParameter() { }
         public string DeploymentParameterType { get { throw null; } set { } }
-        public Azure.ResourceManager.Resources.Models.KeyVaultParameterReference Reference { get { throw null; } set { } }
+        public Azure.ResourceManager.Resources.Models.KeyVaultParameterReferenceAutoGenerated Reference { get { throw null; } set { } }
```

Comment:
@ArcturusZhang could you help understand how to fix this? Based on this comment: https://github.com/Azure/azure-sdk-for-net/pull/49225#discussion_r2034842124